src/widget/wwaveformviewer.cpp

```diff
@@ -77,6 +83,13 @@ void WWaveformViewer::mousePressEvent(QMouseEvent* event) {
         m_pScratchPosition->set(targetPosition);
         m_pScratchPositionEnable->slotSet(1.0);
     } else if (event->button() == Qt::RightButton) {
+        const auto currentTrack = m_waveformWidget->getTrackInfo();
+        auto cueAtClickPos = m_waveformWidget->getCueAtPoint(m_mouseAnchor);
+        if (cueAtClickPos) {
+            m_pCueMenuPopup->setTrackAndCue(currentTrack, *cueAtClickPos);
+            m_pCueMenuPopup->popup(event->globalPos());
+            m_bHotcueMenuShowing = true;
```

Comment:
Why do we need a separate boolean for tracking the status instead of using QWidget::isVisible()?